src/lxc/cgroups/cgfsng.c

```diff
@@ -946,44 +946,47 @@ static void trim(char *s)
 
 static void print_init_debuginfo(struct cgfsng_handler_data *d)
 {
+	struct hierarchy **it;
 	int i;
 
 	if (!getenv("LXC_DEBUG_CGFSNG"))
 		return;
 
-	printf("Cgroup information:\n");
-	printf("  container name: %s\n", d->name);
-	printf("  lxc.cgroup.use: %s\n", cgroup_use ? cgroup_use : "(none)");
-	printf("  lxc.cgroup.pattern: %s\n", d->cgroup_pattern);
-	printf("  cgroup: %s\n", d->container_cgroup ? d->container_cgroup : "(none)");
+	DEBUG("Cgroup information:");
+	DEBUG("  container name: %s", d->name ? d->name : "(null)");
+	DEBUG("  lxc.cgroup.use: %s", cgroup_use ? cgroup_use : "(null)");
+	DEBUG("  lxc.cgroup.pattern: %s", d->cgroup_pattern ? d->cgroup_pattern : "(null)");
+	DEBUG("  cgroup: %s", d->container_cgroup ? d->container_cgroup : "(null)");
 	if (!hierarchies) {
-		printf("  No hierarchies found.\n");
+		printf("  No hierarchies found.");
```

Comment:
Did you mean to use DEBUG there too? if not, why are you dropping the line break?